openquake/baselib/parallel.py

```diff
@@ -811,3 +823,78 @@ class Processmap(BaseStarmap):
     >>> sorted(c.items())
     [('d', 1), ('e', 1), ('h', 1), ('l', 3), ('o', 2), ('r', 1), ('w', 1)]
     """
+
+# ######################## support for grid engine ######################## #
+
+
+def qsub(func, allargs, authkey=None):
+    """
+    Map functions to arguments by means of the Grid Engine.
+
+    :param func: a pickleable callable object
+    :param allargs: a list of tuples of arguments
+    :param authkey: authentication token used to send back the results
+    :returns: an iterable over results of the form (res, etype, mon)
+    """
+    thisfile = os.path.abspath(__file__)
+    host = socket.gethostbyname(socket.gethostname())
+    listener = Listener((host, 0), backlog=5, authkey=authkey)
```

Comment:
This may not work on our cluster (and probably others) when you have multiple interfaces (public and private networks)